src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralCompilingTests.cs

```diff
@@ -0,0 +1,1757 @@
+a
+#endif");
+    }
+
+    [Fact]
+    public void TestPreprocessorConditionInInterpolation()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+$""""""
+{
+#if DEBUG
+42
+#endif
+}
+"""""");",
+                // (4,2): error CS1073: Unexpected token '#'
+                // {
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "").WithArguments("#").WithLocation(4, 2),
+                // (5,1): error CS1003: Syntax error, '}' expected
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_SyntaxError, "#").WithArguments("}").WithLocation(5, 1),
+                // (5,1): error CS1525: Invalid expression term ''
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "#").WithArguments("").WithLocation(5, 1),
+                // (5,1): error CS1056: Unexpected character '#'
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments("#").WithLocation(5, 1),
+                // (7,1): error CS1056: Unexpected character '#'
+                // #endif
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments("#").WithLocation(7, 1));
+    }
+
+    [Fact]
+    public void TestTrivia()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+$""""""
+{
+#if DEBUG
+42
+#endif
+}
+"""""");",
+                // (4,2): error CS1073: Unexpected token '#'
+                // {
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "").WithArguments("#").WithLocation(4, 2),
+                // (5,1): error CS1003: Syntax error, '}' expected
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_SyntaxError, "#").WithArguments("}").WithLocation(5, 1),
+                // (5,1): error CS1525: Invalid expression term ''
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "#").WithArguments("").WithLocation(5, 1),
+                // (5,1): error CS1056: Unexpected character '#'
+                // #if DEBUG
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments("#").WithLocation(5, 1),
+                // (7,1): error CS1056: Unexpected character '#'
+                // #endif
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments("#").WithLocation(7, 1));
+    }
+
+    [Fact]
+    public void TestSingleLineOutput1()
+    {
+        CompileAndVerify(
+@"
+using System;
+
+class C
+{
+    static void Main()
+    {
+        Console.Write($""""""abc""def"""""");
+    }
+}", expectedOutput: @"abc""def");
+    }
+
+    [Fact]
+    public void TestSingleLineOutput2()
+    {
+        CompileAndVerify(
+@"
+using System;
+
+Console.Write($""""""abc""def"""""");
+", expectedOutput: @"abc""def");
+    }
+
+    [Fact]
+    public void TestMultiLineOutput1()
+    {
+        CompileAndVerify(
+@"
+using System;
+
+class C
+{
+    static void Main()
+    {
+        Console.Write($""""""
+                          abc""
```

Comment:
Oof, having these inside regular verbatim strings is a real mind twister!

Reply:
yes :)